### PR TITLE
Add some types for tipsi

### PR DESCRIPTION
**Pull request by henrikra**
I have added TypeScript typings for some functions for this library. I know it is not complete but I used this function so I have added types while developing. I can make more PR's for this in the future when I need more types. I am also sure that this lowers the bar for people to make TS PR's for this awesome library :)

### CODE REVIEW

**Comment by isnifer** [comment]
@henrikra you can see existed types in https://github.com/tipsi/tipsi-stripe/blob/master/src/utils/types.js — please, adopt them all

**Comment by henrikra** [comment]
If somebody wants to add all types to this PR please go ahead! But like I said I can add more types later when I use them. But for now I think best thing to do is merge this even though it is not complete but it helps to get TS typings started :) Does this sound reasonable for you?

**Comment by henrikra** [comment]
@isnifer What is the status of this?

**Comment by isnifer** [comment]
@henrikra so, I have discussed it with my team. We can't merge it without tests.
Basically, I mean — you need to create a new app written on TS. It will use your types and run on Travis CI.

**Comment by isnifer** [comment]
@henrikra that's why we can't even merge this https://github.com/tipsi/tipsi-stripe/pull/263

**Comment by henrikra** [comment]
I don't see a point of doing new app written in TS since these are just types they are not changing any implementation 🤷‍♂️ 

Point of this PR is to add some base for TS types so people can add more types when they see they need them.

What do you think about this?

**Comment by isnifer** [comment]
@henrikra what will be, if we changed an API but forgot to change types?

**Comment by henrikra** [comment]
You have to add to PR template a checkmark that says something like `if I have changed or added the API I have update TypeScript definitions`. :)

**Comment by cybergrind** [comment]
@henrikra we have our internal checklist for PR's but it doesn't save us from errors.
Also, people tend to make mistakes and only tests can prevent this errors from occurring again and again.

I would love to see type annotations but I don't see any point of having type annotations that aren't proved to be working. This isn't maintainable approach and I really don't want to leave team supporting a piece of code that cannot be checked.

If you want to have it merged, please add TS application with tests like it was done in `example`. Otherwise, you can just wait till someone adds it. Soon or later we're going to adopt type annotations for our codebase.

**Comment by orta** [comment]
If you have demo code somewhere in the repo, you can use [dtslint](https://github.com/Microsoft/dtslint) to run the demo code against the types, which means when someone updates the examples it will fail CI

**Comment by isnifer** [comment]
@orta thank you, but we do not have a demo TS code. That's a problem :)

**Comment by peltomaa** [comment]
Any update on this? I would like to use this lib with my Typescript app. If the project maintainers are not interested in maintaining types we should create type file here: https://github.com/DefinitelyTyped/DefinitelyTyped?

**Comment by cybergrind** [comment]
@peltomaa we're definitely not interested in maintaining code without tests. So we're not going to merge this as is.

**Comment by orta** [comment]
Hey all, good points @cybergrind - I've gone and updated the types and added tests to the types in https://github.com/henrikra/tipsi-stripe/pull/1 which would get merged into this pr when @henrikra merges

@yuki24 / @erikdstock want to take the types and completely cover the library, I haven't used the lib and just took some stuff from the docs.

**Comment by yuki24** [comment]
@orta I just took a glance and it seems good to me. @erikdstock?

**Comment by danwoodbury** [comment]
is there any update on this and https://github.com/henrikra/tipsi-stripe/pull/1 being merged? It would be a massive help and I would also be happy to add types while I use them

**Comment by orta** [comment]
As it's been so long, I can switch my PR to be against this repo instead?

**Comment by danwoodbury** [comment]
I think that would be a good idea. I am currently working on integrating apple pay, android pay and card payments into our app so I can add types as I use them if I know they are going to be merged in

**Comment by henrikra** [comment]
I don't still get it that this cannot be merged. I have planted the seed of types which can be extended in later PR's, so what is the problem here? The only problem is that this PR will rotten the longer it takes

**Comment by orta** [comment]
@henrikra - if you merge https://github.com/henrikra/tipsi-stripe/pull/1 then this PR now has tests, which was the unaddressed PR feedback 👍

**Comment by henrikra** [comment]
Lol for some reason I didn't notice that there was PR for my PR 😄

**Comment by henrikra** [comment]
@orta check the CI. Android build failed for `dtslint`

**Comment by hannta** [comment]
It would be nice to get this merged.

**Comment by cybergrind** [comment]
@hannta it needs the fix for https://travis-ci.org/tipsi/tipsi-stripe/jobs/419626026
In general, it is good to go after fix

**Comment by emlynmac** [comment]
Hi, just found this PR after I created my own set of type definitions. These can be found in pr #384 

Looks like this is closer to going in, and follows the JS code a bit more closely in terms of structure, but I've implemented more of the types so maybe some combination of the two would be good?

Regarding the android build error:
npm ERR! missing script: dtslint 
seems to be environmental

File: package.json [hunk 1]
```diff
@@ -37,6 +39,12 @@
     "react-native": ">=0.40"
   },
   "devDependencies": {
+    "dtslint": "^0.3.0",
+    "eslint": "^3.13.1",
+    "eslint-config-tipsi": "1.5.0"
+  },
+  "dependencies": {
+    "prop-types": "^15.5.10",
```
Comment (yassinecc) [inline package.json]: Why is `prop-types` declared twice in the dependencies?

**Comment by henrikra** [comment]
I give up because I am not using this module at the moment

**Pull request closed without merging**